jerry-core/mem/mem-heap.c

```diff
@@ -243,46 +186,48 @@ void *mem_heap_alloc_block_internal (const size_t size)
   const size_t required_size = ((size + MEM_ALIGNMENT - 1) / MEM_ALIGNMENT) * MEM_ALIGNMENT;
   mem_heap_free_t *data_space_p = NULL;
 
-  VALGRIND_DEFINED_SPACE (&mem_heap.first, sizeof (mem_heap_free_t));
+  mem_heap_t *mem_heap = jerry_port_get_heap ();
```

Comment:
Technically, inlining should help (of course we need to prove it, I don't trust the compiler that much as in the past).

My main concern is actually not technical. I don't really understand how this function provides multiple heaps:

    mem_heap_t *jerry_port_get_heap(void)

There is no argument which tells which heap should be used. This patch adds some kind of context management API, but I feel it focuses only on one use case. Since we are in the middle of redesigning the API, we should carefully design this part as well.

What about an OpenGL like jerry_port_set_context()? This would update the appropriate global variables. Or maybe we could add an opaque context pointer, which would be nothing on embedded systems, but would be a pointer on others.

    #if EMBEDDED
    #define CONTEXT
    #else
    #define CONTEXT context *ctx,
    #endif

    retval some_jerry_function(CONTEXT
                                                arg1, ...

I know this would be quite a major change, but provides the most freedom of using Jerry API. I would lijke to hear the others opinion as well, @LaszloLango @akiss77 ?